arbnode/dataposter/slice/slicestorage.go

```diff
@@ -50,40 +52,40 @@ func (s *Storage[Item]) Prune(_ context.Context, until uint64) error {
 	return nil
 }
 
-func (s *Storage[Item]) Put(_ context.Context, index uint64, prevItem *Item, newItem *Item) error {
-	if newItem == nil {
+func (s *Storage) Put(_ context.Context, index uint64, prev, new *storage.QueuedTransaction) error {
+	if new == nil {
 		return fmt.Errorf("tried to insert nil item at index %v", index)
 	}
 	if len(s.queue) == 0 {
-		if prevItem != nil {
+		if prev != nil {
 			return errors.New("prevItem isn't nil but queue is empty")
 		}
-		s.queue = append(s.queue, newItem)
+		s.queue = append(s.queue, new)
 		s.firstNonce = index
 	} else if index == s.firstNonce+uint64(len(s.queue)) {
-		if prevItem != nil {
+		if prev != nil {
 			return errors.New("prevItem isn't nil but item is just after end of queue")
 		}
-		s.queue = append(s.queue, newItem)
+		s.queue = append(s.queue, new)
 	} else if index >= s.firstNonce {
 		queueIdx := int(index - s.firstNonce)
 		if queueIdx > len(s.queue) {
 			return fmt.Errorf("attempted to set out-of-bounds index %v in queue starting at %v of length %v", index, s.firstNonce, len(s.queue))
 		}
-		if !reflect.DeepEqual(prevItem, s.queue[queueIdx]) {
-			return fmt.Errorf("replacing different item than expected at index: %v: %v %v", index, prevItem, s.queue[queueIdx])
+		if !reflect.DeepEqual(prev, s.queue[queueIdx]) {
```

Comment:
I'd feel a bit safer with an Equals method on QueuedTransaction as opposed to using reflect.DeepEqual, now that that's an option. E.g. two of geth's `*types.Transaction`s might differ according to reflect.DeepEqual based on if their hash cache is populated or not. We should instead compare the transactions' hashes in a manual Equals implementation.

Reply:
Done.